src/invocation/ClusterService.ts

```diff
@@ -72,26 +72,27 @@ class ClusterService {
         , attemptPeriod: number
         , deferred: Q.Deferred<ClusterService>) {
         setImmediate(() => {
-            var currentAddress = this.addresses[index];
-            this.client.getConnectionManager().getOrConnect(currentAddress).then((connection: ClientConnection) => {
-                this.ownerConnection = connection;
-                this.initMemberShipListener().then(() => {
-                    deferred.resolve(this);
-                });
-            }).catch((e) => {
-                console.log(e);
-                if (index === this.addresses.length) {
-                    attemptLimit = attemptLimit - 1;
-                    if (attemptLimit === 0) {
-                        var error = new Error('Unable to connect to any of the following addresses ' + this.addresses);
-                        deferred.reject(error);
-                        return;
-                    } else {
-                        setTimeout(this.tryAddressIndex(0, attemptLimit, attemptPeriod, deferred), attemptPeriod);
-                    }
+            if (this.addresses.length <= index) {
```

Comment:
I cannot remember if this was fixing anything :D . Maybe cosmetic changes. I find this one more readable. AttempLimit logic is in `if` part and iterating over connections is in `else` part separately. These were a little mixed before.